### ISSUE TITLE
Upgrading vue-router from 3.2, to 3.4 or 3.5

### ISSUE THREAD

**Issue by eyedean** [quasarframework/quasar#8739]
**Describe the bug**
I am using the [Navigation Failures](https://router.vuejs.org/guide/advanced/navigation-failures.html), which is a Vue-Router 3.4 feature, in my project.  However, @quasar/app is on  a **fixed** version of 3.2.0 as of the [package.json](https://github.com/quasarframework/quasar/blob/dev/app/package.json#L109).  This mismatch causes Typescript to complain because the interface type of `VueRouter` is different in 3.2 vs 3.4.

This is the `tsc` error I am seeing:
```
node_modules/vue-router/types/vue.d.ts:17:5 - error TS2717: Subsequent property declarations must have the same type.  Property 'router' must be of type 'VueRouter | undefined', but here has type 'VueRouter | undefined'.

17     router?: VueRouter
       ~~~~~~

  node_modules/@quasar/app/node_modules/vue-router/types/vue.d.ts:17:5
    17     router?: VueRouter
           ~~~~~~
    'router' was also declared here.
```

**Codepen/jsFiddle/Codesandbox (required)**
Fork a Codepen (https://codepen.io/rstoenescu/pen/xxEvBLR?editors=101) or a jsFiddle (https://jsfiddle.net/rstoenescu/a2cuzods/) or a Codesandbox (https://codesandbox.io/s/optimistic-poitras-4t3m3) and hit save then copy-paste link here.

**To Reproduce**
1. Add VueRouter 3.4 to your project.
2. run `tsc .`

**Expected behavior**
No TypeScript Error.

**Platform (please complete the following information):**
Quasar Version:
@quasar/app Version:
Quasar mode:
  - [X] SPA
  - [X] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

Tested on:
  - [X] SPA
  - [X] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

OS:
Node:
NPM:
Yarn:
Browsers:
iOS:
Android:
Electron:

**Additional context**
I first shared the problem in https://github.com/quasarframework/quasar/issues/8689#issuecomment-811515437 and @IlCallo recommended opening a new issue for it: https://github.com/quasarframework/quasar/issues/8689#issuecomment-811790498

**Comment by eyedean** [comment]
I believe the latest touch on this was for https://github.com/quasarframework/quasar/issues/7152#issuecomment-638748836 that vue-router 3.3 introduced some issues.  The downgrade to vue-router 3.2 has happened 10 months ago and maybe it's time to upgrade it again, if the problems are fixed on their side.

My answer in stackoverflow on similar problem might also be helpful: https://stackoverflow.com/a/65326844

**Comment by rstoenescu** [comment]
Seems like the vue-router bug (see [ticket](https://github.com/quasarframework/quasar/issues/7152#issuecomment-638748836)) is fixed in latest vue-router (3.5.1) so upgraded to it.
Will be available in "@quasar/app" v2.2.7

**Comment by eyedean** [comment]
:tada: Thanks @rstoenescu!
